### PR TITLE
Stac 12961: Add health api to core agent and batcher

### PR DESCRIPTION
**Pull request by craffit**
### Step 1: Link to Jira issue
 Stac 12961

### Step 2: Description of changes

- Add python bindings for the health api
- Add handling of health api to the batcher
- Add molecule test using the integrations_sampling check

### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: pkg/batcher/batch_builder.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+func (builder *BatchBuilder) getHealth(checkID check.ID, stream health.Stream) health.Health {
+	state := builder.getState(checkID)
+
+	if value, ok := state.Health[stream.GoString()]; ok {
+		return value
+	}
+
+	builder.states[checkID] = CheckInstanceBatchState{
+		Topology: state.Topology,
+		Health:   make(map[string]health.Health),
```
Comment (TizianoPerrucci) [inline pkg/batcher/batch_builder.go:78]: here you are re-initializing the health, but what if you have other healths already ?

File: pkg/batcher/batch_builder.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+// NewBatchBuilder constructs a BatchBuilder
+func NewBatchBuilder(maxCapacity int) BatchBuilder {
+	return BatchBuilder{
+		states:       make(map[check.ID]CheckInstanceBatchState),
+		elementCount: 0,
+		maxCapacity:  maxCapacity,
+	}
+}
+
+func (builder *BatchBuilder) getState(checkID check.ID) CheckInstanceBatchState {
```
Comment (TizianoPerrucci) [inline pkg/batcher/batch_builder.go:36]: rename to `getOrCreate` ?

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

File: pkg/collector/python/health_api.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// NOTE
+// Beware that any changes made here MUST be reflected also in the test implementation
+// rtloader/test/health/health.go
+
+// SubmitHealthCheckData is the method exposed to Python scripts to submit health check data
+//export SubmitHealthCheckData
+func SubmitHealthCheckData(id *C.char, healthStream *C.health_stream_t, data *C.char) {
+	goCheckID := C.GoString(id)
+	_stream := convertStream(healthStream)
+	_json := yamlDataToJSON(data)
```
[the remark this answers is not in the document]
Comment (craffit) [reply pkg/collector/python/health_api.go:34]: done

File: pkg/batcher/batcher.go [hunk 4]
```diff
@@ -128,15 +165,23 @@ func (batcher *AsynchronousBatcher) run() {
...
 		switch submission := s.(type) {
 		case submitComponent:
-			batcher.sendTopology(batcher.builder.AddComponent(submission.checkID, submission.instance, submission.component))
+			batcher.sendState(batcher.builder.AddComponent(submission.checkID, submission.instance, submission.component))
 		case submitRelation:
-			batcher.sendTopology(batcher.builder.AddRelation(submission.checkID, submission.instance, submission.relation))
+			batcher.sendState(batcher.builder.AddRelation(submission.checkID, submission.instance, submission.relation))
 		case submitStartSnapshot:
-			batcher.sendTopology(batcher.builder.StartSnapshot(submission.checkID, submission.instance))
+			batcher.sendState(batcher.builder.StartSnapshot(submission.checkID, submission.instance))
```
[the remark this answers is not in the document]
Comment (craffit) [reply pkg/batcher/batcher.go:172]: done

File: pkg/batcher/batcher.go [hunk 5]
```diff
@@ -128,15 +165,23 @@ func (batcher *AsynchronousBatcher) run() {
...
 		case submitStopSnapshot:
-			batcher.sendTopology(batcher.builder.StopSnapshot(submission.checkID, submission.instance))
+			batcher.sendState(batcher.builder.StopSnapshot(submission.checkID, submission.instance))
```
[the remark this answers is not in the document]
Comment (craffit) [reply pkg/batcher/batcher.go:174]: done

File: pkg/batcher/batcher.go [hunk 6]
```diff
@@ -183,6 +228,36 @@ func (batcher AsynchronousBatcher) SubmitStopSnapshot(checkID check.ID, instance
 	}
 }
 
+// SubmitHealthCheckData submits a Health check data record to the batch
+func (batcher AsynchronousBatcher) SubmitHealthCheckData(checkID check.ID, stream health.Stream, data health.CheckData) {
+	log.Debugf("Submitting Health check data for check [%s] stream [%s]: %s", checkID, stream.GoString(), data.JSONString())
```
[the remark this answers is not in the document]
Comment (craffit) [reply pkg/batcher/batcher.go:233]: done

File: pkg/batcher/batcher.go [hunk 7]
```diff
@@ -183,6 +228,36 @@ func (batcher AsynchronousBatcher) SubmitStopSnapshot(checkID check.ID, instance
...
+		checkID:         checkID,
+		stream:          stream,
+		intervalSeconds: intervalSeconds,
+		expirySeconds:   expirySeconds,
+	}
+}
+
+// SubmitHealthStopSnapshot submits a stop of a Health snapshot. This always causes a flush of the data downstream
+func (batcher AsynchronousBatcher) SubmitHealthStopSnapshot(checkID check.ID, stream health.Stream) {
+	log.Debugf("Submitting stop Health snapshot for check [%s] stream [%s]", checkID, stream.GoString())
```
[the remark this answers is not in the document]
Comment (craffit) [reply pkg/batcher/batcher.go:254]: done

File: pkg/batcher/batch_builder.go (hunk 1 above)
Comment (craffit) [reply pkg/batcher/batch_builder.go:78]: done

**Review by TizianoPerrucci — APPROVED** [review]

**Pull request merged**